pyprocessing/math.py

```diff
@@ -162,191 +179,214 @@ def from_angle(angle, target=None):
+
+        if isinstance(other, Iterable):
             return self.dot(other)
-        elif isinstance(other, (list, tuple)):
-            return self.dot(PVector(*other))
-        elif isinstance(other, (float, int)):
-            return self.mult(other)
+
         raise TypeError(
             f"Invalid operand '*' for type PVector and {type(other)}"
         )
 
     def __truediv__(self, other):
         if isinstance(other, (float, int)):
-            return self.div(other)
-        raise TypeError(
-            f"Invalid operand '/' for type PVector and {type(other)}"
-        )
+            return PVector(*(component_a / other for component_a in self))
 
-    def __div__(self, other):
-        if isinstance(other, (float, int)):
-            return self.div(other)
         raise TypeError(
             f"Invalid operand '/' for type PVector and {type(other)}"
         )
 
     def __floordiv__(self, other):
         if isinstance(other, (float, int)):
-            return self.div(other)
+            return PVector(*(component_a // other for component_a in self))
+
         raise TypeError(
             f"Invalid operand '//' for type PVector and {type(other)}"
         )
 
     def __matmul__(self, other):
-        if isinstance(other, PVector):
+        if isinstance(other, Iterable):
             return self.cross(other)
-        raise TypeError(
-            f"Invalid operand '@' for type PVector and {type(other)}"
-        )
 
     def __str__(self):
-        if self.is_2d:
-            return f'{self.x} {self.y}'
-        return f'{self.x} {self.y} {self.z}'
+        return ' '.join(str(i) for i in self)
 
     def __repr__(self):
-        return f'PVector<{", ".join(str(self).split())}> at {id(self)}'
+        return f"PVector<{', '.join(str(i) for i in self)}> at {id(self)}"
+
+    def __len__(self):
+        return len(self.seq)
+
+    def __iter__(self):
+        return iter(self.seq)
 
     def __eq__(self, other):
-        if isinstance(other, PVector):
-            return all(
-                m1 == m2 for m1, m2 in zip(self.position, other.position)
-            )
-        if isinstance(other, (list, tuple)):
-            return all(m1 == m2 for m1, m2 in zip(self.position, other))
-        raise ValueError(
-            f"Invalid operand '==' for type Pvector and {type(other)}"
-        )
+        if not isinstance(other, Iterable):
+            return False
+
+        if not len(other) == len(self):
+            return False
+
+        if not all(component_a == component_b for
+                   component_a, component_b in zip(self, other)):
+            return False
+
+        return True
 
     def __getitem__(self, item):
-        if isinstance(item, str):
-            return {'x': self.x, 'y': self.y, 'z': self.z}[item]
-        if isinstance(item, int):
-            return [self.x, self.y, self.z][item]
-        raise KeyError(f"Item '{item}' cannot be accessed.")
+        return self.seq[item]
```

Comment:
Don't we lose the ability to access x, y, z, w etc by doing `self['x']` ?

If so, that's a regression of the previous code